processor/src/main/java/org/mapstruct/ap/internal/model/source/MethodMatcher.java

```diff
@@ -374,7 +374,7 @@ else if ( typeFromCandidateMethodTypeParameter.isWildCardBoundByTypeVar()
                             // something went wrong
                             return null;
                         }
-                        typeArgs[i] = matchingType.getTypeMirror();
+                        typeArgs[i] = matchingType.getBoxedEquivalent().getTypeMirror();
```

Comment:
I guess this works because if this is not a boxed type it will return type. Perhaps add a comment here. And the javadoc on  `getBoxedEquivalent` should be adapted 😄

Reply:
Perhaps there are more occasions this situation arises. I believe that I somehow covered the scenario where the target is a generic parameter itself in the testcases. Lets just wait and see if more pop up.

Reply:
The Javadoc says:

>Gets the boxed equivalent type if the type is primitive, int will return Integer

What more do you think I should add?

I'll add a comment on the line